### DISCUSSION TITLE
The `deltamethod()` and its future and alternative use in this package?

### DISCUSSION THREAD

**Discussion by stefgehrig** [vincentarelbundock/marginaleffects#1218] [General]
In the [NEWS](https://cran.r-project.org/web/packages/marginaleffects/news/news.html) for the update from 0.17.0 to 0.18.0, it said:

> deltamethod() has been named hypotheses() for a while. We now issue a deprecation warning and it will be removed eventually.

In fact, I have been using the function heavily, because I don't see how else I can **conventiently** (!) get delta method confidence intervals for derived quantities as computed in this example:

```
theta_att_se_dm <- deltamethod(m, 
                            FUN = function(m){
                              p <- predict(m, type = "response", newdata = grid)

                              delta <- grid %>%
                                mutate(p = p) %>%
                                group_by(post, planQI) %>%
                                summarise(mean_p = mean(p), .groups = "drop_last") %>%
                                summarise(diff = diff(qlogis(mean_p)), .groups = "drop") %>%
                                summarise(delta = diff(diff)) %>%
                                pull(delta)
                              
                              theta_att <- grid %>% 
                                mutate(p = p) %>%
                                mutate(p_cf = plogis(qlogis(p)-delta)) %>% 
                                filter(post, planQI) %>% 
                                summarise(theta_att = weighted.mean(p, n) / weighted.mean(p_cf, n)) %>%
                                pull(theta_att)
                              
                              return(theta_att)
                            })
```

I have checked the results against other simulations and they seem fine.
It seems that other delta method implementations, as in `msm` or `car` cannot be used as conveniently as in above code?
Yet, above code does not work anymore since v0.18.0 of this package. What is the reason for abonding the function? Is there an easy substitute in the newer versions of the package to do above operation?

**Comment by vincentarelbundock** [comment]
Have you tried renaming `deltamethod()` to `hypotheses()` in your existing code?

I expect that simple shift to work. If it doesn't, you can file an issue with a minimal reproducible example. Here are some tips on crafting such an example: https://marginaleffects.com/vignettes/help.html#getting-help

**Reply by vincentarelbundock** [reply]
Oh, and `FUN` might have to be `hypothesis` argument

**Reply by stefgehrig** [reply]
Thank you Vincent, that worked. The function's argument is enforced to be called x. So this gives the same result as earlier versions:

`hypotheses(m, hypothesis = function(x) {p <- predict(x, ...})`

with rest as above. Finally, I can update (:
